dotnet/src/Connectors/Connectors.Memory.MongoDB/MongoDBGenericDataModelMapper.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.VectorData;
+using MongoDB.Bson;
+
+namespace Microsoft.SemanticKernel.Connectors.MongoDB;
+
+/// <summary>
+/// A mapper that maps between the generic Semantic Kernel data model and the model that the data is stored under, within MongoDB.
+/// </summary>
+internal sealed class MongoDBGenericDataModelMapper : IVectorStoreRecordMapper<VectorStoreGenericDataModel<string>, BsonDocument>
```

Comment:
Since we aren't taking a dependency from one package to the other, would it make sense to move this into internalutilities?  I'm assuming it's the same in both.

Reply:
Makes sense, added generic data model mapper, default mapper and constants to internal utilities.